File: modules/runtime-repository/src/main/java/org/eclipse/dirigible/runtime/repository/json/Registry.java
/*
 * Copyright (c) 2017 SAP and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * Contributors:
 * SAP - initial API and implementation
 */

package org.eclipse.dirigible.runtime.repository.json;

import java.util.ArrayList;
import java.util.List;

public class Registry extends Collection {

	private static final String TYPE_REGISTRY = "registry";

	private String name;

	private String path;

	private static String type = TYPE_REGISTRY;

	private List<Collection> collections = new ArrayList<Collection>();

	private List<Resource> resources = new ArrayList<Resource>();

	@Override
	public String getName() {
		return name;
	}

	@Override
	public void setName(String name) {
		this.name = name;
	}

	@Override
	public String getPath() {
		return path;
	}

	@Override
	public void setPath(String path) {
		this.path = path;
	}

	@Override
	public List<Collection> getCollections() {
		return collections;
	}

	@Override
	public void setCollections(List<Collection> collections) {
		this.collections = collections;
	}

	@Override
	public List<Resource> getResources() {
		return resources;
	}

	@Override
	public void setResources(List<Resource> resources) {
		this.resources = resources;
	}

	@Override
	public String getType() {
		return type;
	}

}

File: components/ui/service-registry/src/main/resources/META-INF/dirigible/service-registry/registry.js
/*
 * Copyright (c) 2025 Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: Eclipse Dirigible contributors
 * SPDX-License-Identifier: EPL-2.0
 */
angular.module('RegistryService', []).provider('RegistryService', function registryServiceProvider() {
    this.registryServiceUrl = '/services/core/registry';
    this.registryPublicServiceUrl = '/services/core/repository/registry/public'
    this.$get = ['$http', function registryApiFactory($http) {
        /**
         * Loads file content.
         * @param {string} resourcePath - Full resource path.
         */
        const loadContent = function (resourcePath) {
            const url = UriBuilder().path(this.registryPublicServiceUrl.split('/')).path(resourcePath.split('/')).build();
            return $http.get(url);
        }.bind(this);

        /**
         * List the contents of a registry path.
         * @param {string} resourcePath - Full resource path. Default is '/'
         */
        const loadRegistry = function (resourcePath = '/') {
            const url = UriBuilder().path(this.registryServiceUrl.split('/')).path(resourcePath.split('/')).build();
            return $http.get(url, { headers: { 'describe': 'application/json' } });
        }.bind(this);

        return {
            loadContent: loadContent,
            load: loadRegistry,
        };
    }];
});